Teapot/src/Renderer/Model.cpp

```
@@ -7,30 +7,20 @@ namespace Teapot
+        , shapeType(ShapeObjects::PathObj)
     {
         m_HasTexture = true;
         LoadModel(path);
     }
 
-    Model::Model(std::vector<glm::vec3>& positions, std::vector<glm::vec3>& colors, std::vector<glm::vec3>& normals, std::vector<GLuint>& indices, const std::string& nameObject)
+    Model::Model(Shapes::Shape& shapes, const std::string& nameObject, const ShapeObjects type)
         : name(nameObject)
+        , shapeType(type)
     {
+        modelColor = shapes.colors[0];
```

cppcheck (2.10-2)

[performance] Variable 'modelColor' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)